internal/provider/data_source_certificate.go

```diff
@@ -139,3 +160,33 @@ func dataSourceCertificateRead(d *schema.ResourceData, _ interface{}) error {
 
 	return nil
 }
+
+func fetchPeerCertificatesViaTLS(targetURL *url.URL, shouldVerifyChain bool) ([]*x509.Certificate, error) {
+	conn, err := tls.Dial("tcp", targetURL.Host, &tls.Config{
+		InsecureSkipVerify: !shouldVerifyChain,
+	})
+	if err != nil {
+		return nil, err
```

Comment:
Nit: Might be good to return additional error context here

```suggestion
		return nil, fmt.Errorf("unable to open TLS connection: %w", err)
```